pywbem_mock/_inmemoryrepository.py

```diff
@@ -0,0 +1,319 @@
+            data in this store.
+        """
+
+        assert namespace
+        namespace = namespace.strip('/')
+        self.validate_namespace(namespace)
+        return self._repository[namespace]['classes']
+
+    def get_instance_repo(self, namespace):
+        """
+        Gets the handle for the data CIM instance data store for the namespace
+        provided.
+
+          Parameters:
+
+            namespace (:term:`string`):
+              The name of the CIM namespace in the mock repository. Must not be
+              `None`
+          Returns:
+            returns an instance of InMemoryObjStore which defines the
+            methods exists(), get() create(), etc. for accessing the
+            data in this store.
+        """
+
+        assert namespace
+        namespace = namespace.strip('/')
+        self.validate_namespace(namespace)
+        return self._repository[namespace]['instances']
+
+    def get_qualifier_repo(self, namespace):
+        """
+        Gets the handle for the data CIM qualifier declaration data store for
+        the namespace provided.
+
+          Parameters:
+
+            namespace (:term:`string`):
+              The name of the CIM namespace in the mock repository. Must not be
+              `None`
+          Returns:
+            returns an instance of InMemoryObjStore which defines the
+            methods exists(), get() create(), etc. for accessing the
+            data in this store.
+        """
+
+        assert namespace
+        namespace = namespace.strip('/')
+        self.validate_namespace(namespace)
+        return self._repository[namespace]['qualifiers']
+
+    def validate_namespace(self, namespace):
+        """
+        If the namespace exists in the store return.
+
+        Raises:
+
+            TODO
+        """
+
+        namespace = namespace.strip('/')
+        try:
+            self._repository[namespace]
+        except KeyError:
+            raise
+
+    def add_namespace(self, namespace):
+        """
+        """
+
+        assert namespace
+        namespace = namespace.strip('/')
+
+        if namespace in self._repository:
+            raise ValueError
+
+        self._repository[namespace] = {}
+        self._repository[namespace]['classes'] = InMemoryObjStore(
+            True, CIMClass, self._repository)
+
+        self._repository[namespace]['instances'] = InMemoryObjStore(
+            False, CIMInstance, self._repository)
+        self._repository[namespace]['qualifiers'] = InMemoryObjStore(
+            True, CIMQualifierDeclaration, self._repository)
+
+    def remove_namespace(self, namespace):
+        """
+        Remove the namespace from this object
+
+        Raises:
+            KeyError: Namespace not defined in this object store
+        """
+
+        assert namespace
+        namespace = namespace.strip('/')
+
+        if namespace not in self._repository:
+            del namespace
+
+    # TODO: Why not make this a property
```

Comment:
Agreed, it could be a property. Also in the base class. But please double check whether ABCMeta etc support that, e.g. is there an "abstractproperty" decorator?

Reply:
TODO: I have not looked into support for abstractproperty yet